### PR TITLE
`needs_nox` in noxfile

### PR DESCRIPTION
**Pull request by Paulius-Maruska**
Hello, in an attempt to familiarize myself with nox code, I managed to implement #99

### CODE REVIEW

**Comment by theacodes** [comment]
@crwilcox @dhermes @stsewd do any of you have a moment to look at this? I don't have the bandwidth right now.

@Paulius-Maruska thank you for submitting this!

File: nox/_version.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        return None
+
+    # find a "needs_nox" line and parse the version
+    regex = compile(r"""^needs_nox\s*=\s*(['"])(?P<version>[^\1]*)\1.*$""")
+    with open(noxfile, "r") as noxfile:
+        for line in noxfile:
+            m = regex.match(line)
+            if m:
+                needs_version = LooseVersion(m.group("version"))
+                return needs_version
```
Comment (stsewd) [inline nox/_version.py]: Probably this should return the group and create the `LooseVersion` object in the `needs_nox` function

**Comment by Paulius-Maruska** [comment]
I have updated the PR, and also rebased it on the latest master branch.

File: tests/test_tasks.py [hunk 2]
```diff
@@ -64,6 +64,28 @@ def test_load_nox_module_not_found():
...
+
+
+def test_load_nox_module_returns_2_when_needed_version_is_higher():
+    config = _options.options.namespace(
+        noxfile=os.path.join(RESOURCES, "noxfile_future.py")
+    )
+    assert tasks.load_nox_module(config) == 2
+
+
+def test_load_nox_module_returns_2_when_noxfile_does_not_even_import():
```
Comment (crwilcox) [inline tests/test_tasks.py]: It isn't clear to me what this validates. Is this validating the ordering of errors? That we will mention that we have an invalid arg to the decorator before the version?

File: tests/test__version.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+def test_needs_nox_returns_none_when_provided_module_does_not_have_it():
+    class Module:
+        pass
+
+    needed = _v.needs_nox("not/important/noxfile.py", Module)
+    assert needed is None
+
+
+def test_is_version_sufficient_returns_true_for_lower_version():
```
Comment (crwilcox) [inline tests/test__version.py]: should there be a case for exact match?

File: docs/config.rst [hunk 4]
```diff
@@ -313,6 +313,22 @@ Produces these sessions when running ``nox --list``:
     * tests(mysql, new)
 
 
+Minimum required nox version
+----------------------------
+
+Minimum required Nox version can be specified in your ``noxfile.py`` file by simply defining
```
Comment (crwilcox) [inline docs/config.rst]: nitpick: may make sense to drop the word 'simply' as there can be negative feelings about that verbiage in prose.

```suggestion
Minimum required Nox version can be specified in your ``noxfile.py`` file by defining
```

**Comment by theacodes** [comment]
In terms of placating linters and continuing to be somewhat idiomatic, what do y'all think of making `needs_version` a member of the `nox` module?

```python
import nox

nox.needs_version = ">=2018.7.8"
nox.options.sessions = ["lint", "tests-3.6"]

@nox.session
def lint(s):
    ...

...
```

**Comment by crwilcox** [comment]
I think adding this to the module would make sense. It also may be more discoverable via language intelligence services in IDEs. Though my big sticking point is matching PyLint, just to avoid the bug that will follow if this doesn't :D

File: nox/_version.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def is_version_sufficient(noxfile, needed_version):
+    """Return True only when currently running nox version matches or is
+    newer than the version needed.
+    """
+    if needed_version and needed_version > nox_version():
+        logger.error(
+            "Noxfile {} needs at least Nox {} and therefore it cannot be used "
+            "with this version.".format(noxfile, str(needed_version))
```
Comment (stsewd) [inline nox/_version.py]: This message could include the current nox version too

File: nox/_version.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    regex = compile(r"""^needs_nox\s*=\s*(['"])(?P<version>[^\1]*)\1.*$""")
+    with open(noxfile, "r") as noxfile:
+        for line in noxfile:
+            m = regex.match(line)
+            if m:
+                needs_version = m.group("version")
+                return needs_version
+
+
+def needs_nox(noxfile, noxfile_module=None):
```
Comment (stsewd) [inline nox/_version.py]: Np: this function is doing two things. And only using one arg. If noxfile and noxfile module are given, noxfile module is only used

**Comment by Paulius-Maruska** [comment]
I'm sorry for the silence this past week - I was attending the EuroPython2019 conference and was quite busy with the various activities here. :)

I will try to address the issues / respond next week as early as I can.

**Comment by theacodes** [comment]
Take your time, and thanks for keeping us updated. 🙂

**Comment by theacodes** [comment]
I'm going to close this PR due to inactivity. That's not to say I'm not interested in merging it! If you're still up for it just reply and I'll re-open the PR. :)

**Pull request closed without merging**